blockchain/blockdao/blockdao.go

```diff
@@ -142,8 +148,13 @@ func (dao *blockDAO) Start(ctx context.Context) error {
 }
 
 func (dao *blockDAO) checkIndexers(ctx context.Context) error {
-	checker := NewBlockIndexerChecker(dao)
+	var checker *BlockIndexerChecker
 	for i, indexer := range dao.indexers {
+		if _, ok := indexer.(BlockIndexerStateDBWriter); ok {
+			checker = NewBlockIndexerChecker(dao, dao.validator)
```

Comment:
why do we need to validate every block again in the indexers?

Reply:
re-validation should only be required for the fullsync scenario

Reply:
added flag to do this only for fullsync mode/check